backend/src/model/user.ts

```diff
@@ -56,26 +63,61 @@ const findOrCreateUserByGitHubUser = (gitHubUser: GitHubUserType): UserType => {
   return match
 }
 
-interface RegisterProps {
-  username: string
-  email: string
-  password: string
-}
-
 const registerUser = async ({
   username,
   email,
   password,
-}: RegisterProps): Promise<LocalUser> => {
+}: RegisterUserInput): Promise<UserType|null> => {
   const queryText =
     'INSERT INTO USERS(username, email, password) VALUES($1, $2, $3) RETURNING user_id, username, email'
   const cryptedPassword = bcrypt.hashSync(password, 10)
-  const queryResult = await pool.query<LocalUser>(queryText, [
+  const queryResult = await pool.query<UserRecord>(queryText, [
     username,
     email,
     cryptedPassword,
   ])
-  return queryResult.rows[0]
+
+  if (queryResult.rows.length === 0) {
+    return null
+  }
+
+  const user = {
+    id: queryResult.rows[0].user_id,
+    username: queryResult.rows[0].username,
+    email: queryResult.rows[0].email,
+  }
+
+  return user
+}
+
+const findUserByUsername = async (username: string): Promise<UserType|null> => {
+  const queryText =
+    'SELECT * FROM USERS WHERE username = $1'
+  const queryResult = await pool.query<UserRecord>(queryText, [
+    username,
+  ])
+  
+  if (queryResult.rows.length === 0) {
+    return null
+  }
+
+  const user = {
+    id: queryResult.rows[0].user_id,
+    username: queryResult.rows[0].username,
+    email: queryResult.rows[0].email,
+    password: queryResult.rows[0].password,
+  }
+
```

Comment:
This object could be created inside the sql query by selecting right fields.

Reply:
to avoid assigning names in sql queries perhaps better to change the column names in db to be as they are in objects?

Reply:
postgres doesn't like camelCase in column names so we "have" to use snake_case in those. And if we change every table id to be id for example `user_id => id`, I think this doesn't work in migration `user_id INTEGER REFERENCES USERS ON DELETE CASCADE,`

Reply:
It works, just needed to change:

`
CREATE TABLE USERS(
    id serial PRIMARY KEY,
....
`    
and when referencing

`
CREATE TABLE SERVICE_USERS(
    id serial PRIMARY KEY,
    user_id INTEGER REFERENCES USERS(id) ON DELETE CASCADE,
`

Reply:
ok nice. Le'ts do the change then